lib/scrapPaperReader.js

```diff
@@ -10,4 +85,410 @@ const readPdf = async (pdfBuffer) => {
+    );
+  }
+  return jpegs;
+};
+
+/**
+ * Detects and extracts a barcode on a jpeg file
+ * We probably want to start working on some image analysis that helps read
+ * barcodes based on uploaded scans. Many likely will not read. This will require
+ * some optimization here.
+ * @param {string} jpeg jpeg file object from arr produced by _convertPdf()
+ * @returns {string} a code-128 formatted barcode or undefined if not found
+ */
+const _decodeJpeg = async (jpeg, logger) => {
+  // priortize most effective patchSizes first
+  const patchSizes = ['medium', 'small', 'x-small', 'large', 'x-large'];
+
+  if (!jpeg || typeof jpeg.pageNum != 'number' || !jpeg.jpegFilepath || !jpeg.workingDir) {
+    throw Error('Invalid jpeg file or missing metadata');
+  }
+  logger.info('_decodeJpeg() starting new jpeg', jpeg.jpegFilepath);
+
+  let barcode = null;
+
+  for (let i = 0; i < segmentOptions.length; i++) {
+    const segmentFilepaths = await _segmentJpeg(
+      jpeg.pageNum,
+      jpeg.jpegFilepath,
+      jpeg.workingDir,
+      logger,
+      i
+    );
+    for (let i = 0; i < patchSizes.length; i++) {
+      for (let k = 0; k < segmentFilepaths.length; k++) {
+        // break to avoid additional decoding and image segmentation if found
+        barcode = await _decodeJpegSegment(segmentFilepaths[k], patchSizes[i]);
+        if (barcode) {
+          logger.info(
+            `_decodeJpeg() found barcode, ${barcode}, with segmentation option ${k} and framesize '${patchSizes[i]}'`
+          );
+          return barcode;
+        }
+      }
+    }
+  }
+  return barcode;
+};
+
+/**
+ * Quagga fails to locate barcodes on the borders of full-sized pages because:
+ * (1.) barcode appears almost too small for x-small patchSize option to read unless super high resolution
+ * (2.) the locator fails to locate barcodes on edges/borders of a large image. It generally looks in the center.
+ * Hence, we segment a full jpeg and input segments so barcode appears in center of segment and barcode appears bigger
+ * relative to patch size.
+ * @param {string} filepath vertical slice of a jpeg full page
+ * @returns {string} a barcode string or null
+ */
+const _decodeJpegSegment = async (filepath, patchSize) => {
+  return new Promise((resolve, reject) => {
+    try {
+      // as barcode changes size relative to page, we should adjust patchSize. May want to guarantee barcode size and page size.
+      quagga.decodeSingle(
+        {
+          src: filepath,
+          halfSample: true,
+          locate: true,
+          patchSize: patchSize,
+          numOfWorkers: 0,
+          inputStream: {
+            size: 1920, // restrict input-size resolution to be 1920 in width (long-side)
+          },
+          decoder: {
+            readers: ['code_128_reader'],
+          },
+        },
+        (result) => {
+          if (result && result.codeResult && result.codeResult.code) {
```

Comment:
Nit, optional chaining can make this a little more legible:
```suggestion
          if (result?.codeResult?.code) {
```